Add configuration profile options

rclee opened the pull request:
- Creates new class MMEEventsService to manage the creation and distribution of MMEEventsConfiguration objects/configurations.
- Modifies MMELocationManager and MMEEventsManager to use MMEEventsService configuration
- Adds and repairs cedar tests
- Fix bug with geofence creation related to custom profile
- Update namespace dependencies symbols

rclee committed:
Port Radius Profile Configuration (#36)

* Switch to @available for version checking

This requires Xcode 9.x.

* Port Radius Profile Configuration

- Creates new class MMEEventsService to manage the creation and distribution of MMEEventsConfiguration objects/configurations.
- Expands MMEEventsConfiguration to include a variable hibernation radius configuration and potentially additional configurations in the future. A user may access the variable radius by setting both "MMEEventsProfile" (with the appropriate key "VariableGeofence") and "MMEGeofenceRadius" with a desired radius (within a range we've set) in the applications plist. If "MMEEventsProfile" is set in plist without choosing a radius then a median radius will be the default. 
- Modifies MMELocationManager and MMEEventsManager to use MMEEventsService configuration
- Adds and repairs cedar tests

* Fix bug with geofence creation related to custom profile

geofence were being created too often when the profile was set to a wide geofence radius.

rclee committed:
Update symbols

fabian-guerra commented on MapboxMobileEvents/MMEEventsManager.m:
```diff
@@ -54,7 +55,7 @@ - (instancetype)init {
         _eventQueue = [NSMutableArray array];
         _commonEventData = [[MMECommonEventData alloc] init];
-        _configuration = [MMEEventsConfiguration defaultEventsConfiguration];
+        _configuration = [MMEEventsService.sharedService configuration];
```
Let's keep the bracket convention  `[[MMEEventsService sharedService] configuration]`

fabian-guerra reviewed (commented):
Overall looks good. Just minor changes. Have you tested energy consumption with this change?

rclee committed:
Fix nullable singleton

rclee commented:
@fabian-guerra Yes, battery testing has been done. Results were mostly as expected with the increase in geofence radius.

akitchen commented:
LGTM. What's up with CI? Can you fix up merge conflicts and re-push before final 👍  ?

rclee committed:
Merge branch 'master' into develop

rclee committed:
Follow bracket convention

rclee commented on MapboxMobileEvents/MMELocationManager.m:
```diff
@@ -30,6 +33,7 @@ - (instancetype)init {
         _application = [[MMEUIApplicationWrapper alloc] init];
         NSArray *backgroundModes = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIBackgroundModes"];
         _hostAppHasBackgroundCapability = [backgroundModes containsObject:@"location"];
+        _configuration = [MMEEventsService.sharedService configuration];
```
Right, sorry about that. Thanks for catching those.

fabian-guerra reviewed (dismissed):
Thank you. 👍🏼

rclee committed:
Fix race condition in cedar test

"created" NSDate was causing a failure some of the time when the date time was off by a second.

rclee commented on MapboxMobileEventsTests/MMEEventsManagerTests.mm:
```diff
@@ -831,14 +831,21 @@ @interface MMEEventsManager (Tests) <MMELocationManagerDelegate>
             it(@"enqueues the correct event", ^{
                 CLLocation *location = [[CLLocation alloc] initWithLatitude:visit.coordinate.latitude longitude:visit.coordinate.longitude];
-                NSDictionary *attributes = @{MMEEventKeyCreated: [eventsManager.dateWrapper formattedDateStringForDate:[location timestamp]],
+                NSDictionary *attributes = @{MMEEventKeyCreated: @"2018-05-30T21:53:08.720+0000",
```
Right, that was what I was using before and what was causing an issue. When the enqueue event is created with the `didVisit:` method, then a "created" time is added to the dictionary via NSDate. After that the `expectedVisitEvent`'s "created" attribute was added at nearly the same time (via NSDate) but would sometimes be off by a second and cause the test to fail.

fabian-guerra replied:
I was thinking more into a 'static' date you could use on both cases:
```
NSDictionary *attributes = @{MMEEventKeyCreated: [eventsManager.dateWrapper formattedDateStringForDate:[location timestamp]],
```
And
```
[tempDict setObject:[eventsManager.dateWrapper formattedDateStringForDate:[location timestamp]] forKey:@"created"];
```

`[location timestamp]` would be the same in both cases right?

rclee replied:
Yes, you're right...I wasn't putting two and two together there. Thanks!

rclee committed:
Remove hard-coded date string

fabian-guerra reviewed (approved):
👍🏼
